src/shells.ts

```diff
@@ -0,0 +1,30 @@
+import * as vscode from 'vscode';
+
+export enum OS {
+  WINDOWS = 1,
+  LINUX = 2
+}
+
+const windowsDefaultShell = 'powershell.exe';
```

Comment:
@TylerLeonhardt any advice on how best to launch a good "default shell" with powershell? powershell.exe will open Windows PowerShell which is wrong sometimes. I'm thinking something along the lines of `pwsh || powershell` and suppressing any pwsh not existing errors. We're currently connecting to the host using:

```
ssh (host) powershell -NoExit -Command cd (path)\;(shell)`
```

Reply:
@Tyriar i am using powershell as default because is the default of terminal.integrated.shell.windows in vscode if you change it in settings.json it uses you preferred shell.

Reply:
Just so we're clear, @Tyriar you're not really talking about this PR... but rather at the `terminal.integrated.shell.windows` level? Where VS Code could maybe check if `pwsh` is installed and default to that instead?

[This (rather large) file](https://github.com/PowerShell/vscode-powershell/blob/master/src/platform.ts) in vsode-powershell will get you all the available PowerShell executables on any platform. There's a:
```ts
powerShellExeFinder.getFirstAvailablePowerShellInstallation()
```

That will get the first PowerShell that's available on the system.

That file is just a tad specific to the PowerShell extension because it uses [this interface](https://github.com/PowerShell/vscode-powershell/blob/master/src/platform.ts#L10) but beyond that, it'll work.

Reply:
The complications in discovering PowerShell is that now we have:

* MSI install
* MSIX install (aka the Store release)
* dotnet global tool
* ^ those but preview instead of stable
* Windows PowerShell

Reply:
@TylerLeonhardt well it could be in this PR if it's easy, I don't want to use a giant file to find it though. Is there an easy way in a single powershell command to run pwsh.exe if it's available, otherwise powershell.exe (this should always be there?) and not error?

Reply:
Scratch that, what we actually want is a script to run one shell (whatever `terminal.integrated.shell.windows` is set to) if it exists, otherwise fallback to `powershell.exe`. So we still need to create a command to run the shell if the file exists and fallback without error if not.

Reply:
That's what they have in this PR already, yeah?